AK/UnicodeUtils.h

```diff
@@ -12,6 +12,19 @@
 
 namespace AK::UnicodeUtils {
 
+constexpr int bytes_to_store_code_point_in_utf8(u32 code_point)
+{
+    if (code_point <= 0x7f)
+        return 1;
+    if (code_point <= 0x7ff)
+        return 2;
+    if (code_point <= 0xffff)
+        return 3;
+    if (code_point <= 0x10ffff)
+        return 4;
+    return -1;
+}
```

Comment:
This function should return size_t and use an invalid default of 0. Here's why:

For the only user up in `from_code_point`, this would happen without the unicode check: invalid code point -> `bytes_to_store_code_point_in_utf8` returns -1 -> C++ jovially casts this to size_t for `replace_with_new_short_string`, resulting in 2^32 -> short string limit is blown and VERIFY triggers. To me it's conceivable that someone might do something similar with this function in another context, but have no valid code point check, instantly asking the operating system for 4GB of memory or so. An invalid code point causing a 0 will still make all the code behave correctly (or incorrectly, if you pass an invalid code point), but it will no longer trigger UB and cause insane side effects. It also makes logical sense to me to use an invalid default of 0, since you can't store an invalid code point in UTF-8 anywhere.

Reply:
Fine, but there is `VERIFY(is_unicode(code_point));` in the `String::from_code_point` that is a single user of this function.